test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -0,0 +1,208 @@
+        self.assertTrue(res['rc'] == 0)
+
+    def test_capture_server(self):
+        """
+        Test the 'capture_server' interface of PBSSnapUtils
+        """
+
+        # Set something on the server so we can match it later
+        job_hist_duration = "12:00:00"
+        attr_list = {"job_history_enable": "True",
+                     "job_history_duration": job_hist_duration}
+        self.server.manager(MGR_CMD_SET, SERVER, attr_list)
+
+        output_dir = os.path.join("/", "tmp", "snapshot_server")
+        snap_args = {"out_dir": output_dir, "num_service_logs": 2,
+                     "num_acct_logs": 5}
+
+        with PBSSnapUtils(snap_args) as snap_obj:
+            snap_obj.capture_server(True, True)
+
+        # Go through the snapshot and perform certain checks
+        # Check 1: the snapshot exists
+        self.assertTrue(os.path.isdir(output_dir))
+        # Check 2: all directories except the 'server' directory have no files
+        outdir_fullpath = os.path.abspath(output_dir)
+        svr_fullpath = os.path.join(outdir_fullpath, "server")
+        for root, _, files in os.walk(output_dir):
+            for file in files:
+                file_fullpath = os.path.join(root, file)
+                # Find the common paths between 'server' & the file
+                common_path = os.path.commonprefix([file_fullpath,
+                                                    svr_fullpath])
+                self.assertTrue(ntpath.basename(common_path) == "server")
+        # Check 3: qstat_Bf.out exists
+        qstat_bf_path = os.path.join(output_dir, "server", "qstat_Bf.out")
+        self.assertTrue(os.path.isfile(qstat_bf_path))
+        # Check 4: qstat_Bf.out has 'job_history_duration' set to 24:00:00
+        with open(qstat_bf_path, "r") as fd:
+            for line in fd:
+                if "job_history_duration" in line:
+                    # Remove whitespaces
+                    line = "".join(line.split())
+                    # Split it up by '='
+                    key_val = line.split("=")
+                    self.assertTrue(key_val[1] == job_hist_duration)
+
+        # Cleanup
+        self.du.rm(path=outdir_fullpath, recursive=True, force=True)
+
+    def test_capture_all(self):
+        """
+        Test the 'capture_all' interface of PBSSnapUtils
+        Note: this will test out all capture_ interfaces of PBSSnapUtils
+        except capture_pbs_logs
+        """
+
+        output_dir = "snapshot_all"
+        snap_args = {"out_dir": output_dir, "num_service_logs": 2,
+                     "num_acct_logs": 5}
+
+        with PBSSnapUtils(snap_args, create_tar=True) as snap_obj:
+            snap_obj.capture_all()
+
+        # Cleanup
+        self.du.rm(path=output_dir, recursive=True, force=True)
+
+    def test_capture_pbs_logs(self):
+        """
+        Test the 'capture_pbs_logs' interface of PBSSnapUtils
+
+        WARNING: Assumes that the test is being run on type - 1 PBS install
```

Comment:
Our customers usually don't run a mom on the front end node (where the server is running).  This is actually the default for OSS.  You can check for the scheduler and comm if you like.  Those don't really need to be started for pbs_snapshot to work.  You just collect config/log files from them.  The only daemon that has to actually be running is the server.  Whichever way you want to look at this, I'm fine with.